### PR TITLE
Time display component

### LINKED ISSUE #771: implement time display component
### Subject of the issue
Add a time display component which can be later used for various time displays, e.g. on the leaderboard, order history table, etc.

### Deliverables
- Create a clean time display component
- Make sure it switches between 24 and 12h format when clicking on the time
- Show date format according to user locale
- Refactor leaderboard and "Trades" tab to use the new time display component

### Reward
50 SNX

### PR DESCRIPTION
**Pull request by 0xwontonsoup**
## Description
Fixes #771 

## Related issue
#771 📦 

## Motivation and Context
A "DIY"-style refactor with the additional feature of toggling between 12h/24h format when the component is clicked

# Screenshots
https://user-images.githubusercontent.com/99768011/167759392-276c8fdf-10b6-4e66-afea-fb33a85d564d.mov

https://user-images.githubusercontent.com/99768011/167759429-69917ce0-6de7-428a-9129-e697cbf76f0a.mov

### CODE REVIEW

**Comment by 0xwontonsoup** [comment]
> this is looking good so far, can we replace `time` column on the Trades tab with this Timer component?
> 
> https://github.com/Kwenta/kwenta/blob/42c831eda3ac20bbd7dfa27465994af54001b799/sections/futures/Trades/Trades.tsx#L87

weird, i thought i had pushed a change that did that

**Review by brossetti1 — CHANGES_REQUESTED** [review]
so looking into some options in regards to the `getLocale` file, im wondering if we can just use [intlFormat](https://date-fns.org/v2.28.0/docs/intlFormat) here and pass the language to that?

if you have any reservations against that let me know - if we decide to keep the getLocale - it seems like something more broad than `sections/futures/Trades/getLocale.ts` -- maybe we can put this in `/utils/formatters`?

File: sections/futures/Trades/TimeDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	navigator.languages !== undefined
+		? (language.code = navigator.languages[0])
+		: (language.code = navigator.language);
+
+	const date = format(new Date(cellPropsValue), 'MM/dd/yy', {
+		locale: getLocale(language.code),
+	});
+	const time12hr = new Date(cellPropsValue).toLocaleTimeString(language.code);
+	const time24hr = format(new Date(cellPropsValue), 'HH:mm:ss', {
```
Comment (brossetti1) [inline sections/futures/Trades/TimeDisplay.tsx:27]: so looking into some options in regards to the `format`, im wondering if we can just use [intlFormat](https://date-fns.org/v2.28.0/docs/intlFormat) here and pass the locale to that?

**Review by brossetti1 — APPROVED** [review]
not going to hold this up over the width comment

**Pull request merged**